### PR TITLE
JAMES-2366 Fix a bug in getAllMappings

### PR DESCRIPTION
**Pull request by chibenwa**
We were never actually examining the detailed content of this call.

It turned out to be buggy for both HBASE and Cassandra. I fixed it for Cassandra implementation.

### CODE REVIEW

File: server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java [hunk 1]
```diff
@@ -112,6 +113,34 @@ public void testStoreAndRetrieveRegexMapping() throws ErrorMappingException, Rec
         assertThat(virtualUserTable.getAllMappings()).describedAs("No mapping").isNull();
     }
 
+    @Test
+    public void getAllMappingsShouldListAllEntries() throws ErrorMappingException, RecipientRewriteTableException {
+        String user = "test";
+        String user2 = "test2";
+        Domain domain = Domain.LOCALHOST;
+        // String regex = "(.*):{$1}@localhost";
```
Comment (mbaechler) [inline server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java]: ?

File: server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java [hunk 2]
```diff
@@ -112,6 +113,34 @@ public void testStoreAndRetrieveRegexMapping() throws ErrorMappingException, Rec
...
+        // String regex2 = "(.+):{$1}@test";
+        String regex = "(.*)@localhost";
+        String regex2 = "(.+)@test";
+
+        assertThat(virtualUserTable.getMappings(user, domain)).describedAs("No mapping").isNull();
```
Comment (mbaechler) [inline server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java]: you should not assert anything at this point

File: server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java [hunk 3]
```diff
@@ -295,6 +295,22 @@ public void addressGroupsEndpointShouldHandleRequests() throws Exception {
         assertThat(members).containsOnly("user1@domain.com", "user2@domain.com");
     }
 
+    @Test
+    public void addressForwardsEndpointShouldHandleRequests() throws Exception {
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java]: shouldListForwardAddresses ?

File: server/data/data-hbase/src/test/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTableTest.java [hunk 4]
```diff
@@ -60,4 +61,10 @@ protected AbstractRecipientRewriteTable getRecipientRewriteTable() throws Except
         rrt.configure(new DefaultConfigurationBuilder());
         return rrt;
     }
+
+    @Ignore("JAMES-2376 HBASE RRT getAllMappings is buggy")
+    @Override
+    public void getAllMappingsShouldListAllEntries() {
```
Comment (mbaechler) [inline server/data/data-hbase/src/test/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTableTest.java]: can't you fix it quickly ?

Comment (chibenwa) [reply server/data/data-hbase/src/test/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTableTest.java]: I'll give it a go tomorrow in a       separate PR?

Comment (rouazana) [reply server/data/data-hbase/src/test/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTableTest.java]: I did a quick fix. But maybe there is a better way to do that with hbase...

**Comment by chibenwa** [comment]
:+1:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**